### PR TITLE
libfilezilla: 0.41.0 -> 0.45.0; filezilla: 3.63.1 -> 3.66.4

### PR DESCRIPTION
**Pull request by Intuinewin**
###### Description of changes

Sources are now distributed in a .tar.xz

https://filezilla-project.org/changelog.php
https://lib.filezilla-project.org/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libfilezilla/default.nix [hunk 1]
```diff
@@ -31,6 +31,7 @@ stdenv.mkDerivation rec {
     description = "A modern C++ library, offering some basic functionality to build high-performing, platform-independent programs";
     license = licenses.gpl2Plus;
     maintainers = with maintainers; [ pSub ];
-    platforms = platforms.unix;
+    # darwin requires apple_sdk >= 10.13
+    platforms = lib.platforms.linux;
```
Comment (LeSuisse) [inline pkgs/development/libraries/libfilezilla/default.nix:35]: We have `apple_sdk_11_0` so it should be manageable.

File: pkgs/applications/networking/ftp/filezilla/default.nix [hunk 2]
```diff
@@ -19,11 +19,11 @@
 
 stdenv.mkDerivation rec {
   pname = "filezilla";
-  version = "3.63.1";
+  version = "3.64.0";
```
Comment (LeSuisse) [inline pkgs/applications/networking/ftp/filezilla/default.nix:22]: Would you mind updating to 3.66.4+ so we can get the mitigation for CVE-2023-48795?

**Review by LeSuisse — COMMENTED** [review]
Changes look fine, quick testing did not show any issue.

Result of `nixpkgs-review pr 231828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>filezilla</li>
    <li>libfilezilla</li>
  </ul>
</details>

File: pkgs/applications/networking/ftp/filezilla/default.nix (hunk 2 above)
Comment (Intuinewin) [reply pkgs/applications/networking/ftp/filezilla/default.nix]: Done

File: pkgs/development/libraries/libfilezilla/default.nix (hunk 1 above)
Comment (Intuinewin) [reply pkgs/development/libraries/libfilezilla/default.nix]: Indeed, the lib now works but I'm still trying to compile filezilla on macos

**Review by LeSuisse — APPROVED** [review]
Works fine on Linux. Darwin builds on OfBorg look good too so let's merge this.

Thanks for working on it!

Result of `nixpkgs-review pr 231828` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>filezilla</li>
    <li>libfilezilla</li>
  </ul>
</details>

**Comment by LeSuisse** [comment]
3.66.4 contains the mitigation for CVE-2023-48795.

Looking at the version history it should be fine to backport the upgrade.

https://filezilla-project.org/versions.php?version=3.66.4&oldversion=3.63.1

**Pull request merged**